src/connector.rs

```diff
@@ -0,0 +1,558 @@
+use crate::fungible_token::*;
+use crate::parameters::*;
+use crate::sdk;
+use crate::types::*;
+
+use crate::deposit_event::*;
+use crate::json::{parse_json, FAILED_PARSE};
+use crate::prelude::{Address, U256};
+use crate::prover::{validate_eth_address, Proof};
+#[cfg(feature = "log")]
+use alloc::format;
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+pub const CONTRACT_NAME_KEY: &str = "EthConnector";
+pub const CONTRACT_FT_KEY: &str = "EthConnector.ft";
+pub const NO_DEPOSIT: Balance = 0;
+const GAS_FOR_FINISH_DEPOSIT: Gas = 10_000_000_000_000;
+const GAS_FOR_VERIFY_LOG_ENTRY: Gas = 40_000_000_000_000;
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct EthConnectorContract {
+    contract: EthConnector,
+    ft: FungibleToken,
+}
+
+/// eth-connector specific data
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct EthConnector {
+    pub prover_account: AccountId,
+    pub eth_custodian_address: EthAddress,
+}
+
+impl EthConnectorContract {
+    pub fn new() -> Self {
+        Self {
+            contract: sdk::get_contract_data(CONTRACT_NAME_KEY),
+            ft: sdk::get_contract_data(CONTRACT_FT_KEY),
+        }
+    }
+
+    pub fn init_contract() {
+        //assert_eq!(sdk::current_account_id(), sdk::predecessor_account_id());
+        assert!(
+            !sdk::storage_has_key(CONTRACT_NAME_KEY.as_bytes()),
+            "ERR_CONTRACT_INITIALIZED"
+        );
+        #[cfg(feature = "log")]
+        sdk::log("[init contract]".into());
+        let args: InitCallArgs =
+            InitCallArgs::from(parse_json(&sdk::read_input()).expect_utf8(FAILED_PARSE));
+        let current_account_id = sdk::current_account_id();
+        let owner_id = String::from_utf8(current_account_id).unwrap();
+        let mut ft = FungibleToken::new();
+        ft.internal_register_account(&owner_id);
+        let contract_data = EthConnector {
+            prover_account: args.prover_account,
+            eth_custodian_address: validate_eth_address(args.eth_custodian_address),
+        };
+        Self {
+            contract: contract_data,
+            ft,
+        }
+        .save_contract();
+    }
+
+    pub fn deposit_near(&self) {
```

Comment:
This is maybe a silly question, but how does a user inside the EVM use the Near they have deposited?

Reply:
@djsatok